test/TestUtils/TestProjectFixture.cs

```diff
@@ -0,0 +1,362 @@
+                .WorkingDirectory(_testProject.ProjectDirectory)
+                .Environment("NUGET_PACKAGES", _repoDirectoriesProvider.NugetPackages)
+                .CaptureStdErr()
+                .CaptureStdOut()
+                .Execute()
+                .EnsureSuccessful();
+
+            _testProject.LoadOutputFiles();
+
+            return this;
+        }
+
+        public TestProjectFixture RestoreProject(string[] fallbackSources)
+        {
+            var restoreArgs = new List<string>();
+            foreach (var fallbackSource in fallbackSources)
+            {
+                restoreArgs.Add("-f");
+                restoreArgs.Add(fallbackSource);
+            }
+
+            _sdkDotnet.Restore(restoreArgs.ToArray())
+                .WorkingDirectory(_testProject.ProjectDirectory)
+                .CaptureStdErr()
+                .CaptureStdOut()
+                .Environment("NUGET_PACKAGES", _repoDirectoriesProvider.NugetPackages)
+                .Execute()
+                .EnsureSuccessful();
+
+            return this;
+        }
+
+        public TestProjectFixture EnsureRestored(params string[] fallbackSources)
+        {
+            if ( ! _testProject.IsRestored())
+            {
+                RestoreProject(fallbackSources);
+            }
+
+            return this;
+        }
+
+        public TestProjectFixture Copy()
+        {
+            return new TestProjectFixture(this);
+        }
+
+        public TestProjectFixture MoveDepsJsonToSubdirectory()
+        {
+            var subdirectory = Path.Combine(_testProject.ProjectDirectory, "d");
+            if (!Directory.Exists(subdirectory))
+            {
+                Directory.CreateDirectory(subdirectory);
+            }
+
+            var destDepsJson = Path.Combine(subdirectory, Path.GetFileName(_testProject.DepsJson));
+
+            if (File.Exists(destDepsJson))
+            {
+                File.Delete(destDepsJson);
+            }
+            File.Move(_testProject.DepsJson, destDepsJson);
+
+            _testProject.DepsJson = destDepsJson;
+
+            return this;
+        }
+
+        public TestProjectFixture MoveRuntimeConfigToSubdirectory()
+        {
+            var subdirectory = Path.Combine(_testProject.ProjectDirectory, "r");
+            if (!Directory.Exists(subdirectory))
+            {
+                Directory.CreateDirectory(subdirectory);
+            }
+
+            var destRuntimeConfig = Path.Combine(subdirectory, Path.GetFileName(_testProject.RuntimeConfigJson));
+
+            if (File.Exists(destRuntimeConfig))
+            {
+                File.Delete(destRuntimeConfig);
+            }
+            File.Move(_testProject.RuntimeConfigJson, destRuntimeConfig);
+
+            _testProject.RuntimeConfigJson = destRuntimeConfig;
+
+            return this;
+        }
+
+        public TestProjectFixture ReplaceTestProjectOutputHostFromDotnet(DotNetCli dotnet = null)
```

Comment:
Can you explain why we need to do this?  Shouldn't the test app be referencing the locally built DotNetHost packages, and getting these files from those packages as normal?